File: frontend/src/microservices/order/objects/PaymentPlannerResponse.ts
// 本文件定义订单支付响应，和后端 `PaymentPlannerResponse` 保持同名镜像。

export type PaymentPlannerResponse = {
  paymentId: string
  paymentAmount: string
  paymentCurrency: string
  paymentMethod: string
  paymentStatus: string
  authorizedAt: string
  capturedAt: string | null
}

export const paymentPlannerResponseFromJson = (json: string): PaymentPlannerResponse =>
  JSON.parse(json) as PaymentPlannerResponse

export const paymentPlannerResponseToJson = (value: PaymentPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/order/objects/PaymentPlannerResponse.scala
package com.typesafe.travel.order.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class PaymentPlannerResponse(paymentId: String, paymentAmount: String, paymentCurrency: String, paymentMethod: String, paymentStatus: String, authorizedAt: String, capturedAt: Option[String])
object PaymentPlannerResponse:
  given Encoder[PaymentPlannerResponse] = deriveEncoder
  given Decoder[PaymentPlannerResponse] = deriveDecoder
